File: classTask/Level1/level1.py
# number 0
def print_hello_world():
    print("Hello World")


# number 1
def second_largest(*numbers):
    highest = 0
    second_highest = 0
    for number in numbers:
        if number > highest:
            highest = number
    for number in numbers:
        if number == highest:
            continue
        if number > second_highest:
            second_highest = number
    return second_highest


# number 2
def print_semicolon():
    semicolon = "SEMICOLON"
    for letter in semicolon:
        print(letter)


# number 3
def print_ventures():
    ventures = "VENTURES"
    count = 0
    for letter in ventures:
        if count % 2 == 0:
            print()
        print(letter, end=' ')
        count += 1
    print()


# number 4
def find_occurrences():
    word = "Mississippi"

    count_for_s = word.count('s')
    count_for_i = word.count('i')
    print("Number of 's': " + str(count_for_s))
    print("Number of 'i': " + str(count_for_i))


# number 5
def sum_divisible_by_3():
    sum_of_dividers = 0
    for i in range(1, 31):
        if i % 3 == 0:
            sum_of_dividers += i
    return sum_of_dividers


# number 6
def guess_birth_date():
    birthday = 15
    answer = 0
    while answer != birthday:
        answer = int(input("Enter a guess: "))
        if answer != birthday:
            print("Incorrect Guess!")
    print("Correct Guess!")


# number 7
def infinite_series():
    pie = 4.0
    denominator = 3.0
    found = False
    terms = 1

    while not found:
        if terms % 2 == 0:
            pie += 4.0 / denominator
        else:
            pie -= 4.0 / denominator
        denominator += 2.0
        print(f"{terms: <10}{pie:.7f}")
        terms += 1

        if format(pie, '.7f') == '3.1415900':
            found = True

    print("Pie:", pie)
    print("Number of terms:", terms)



if __name__ == "__main__":
    print_hello_world()
    print_semicolon()
    print_ventures()
    print(second_largest(2, 4, 70, 12, 55, 90, 532, 45, 12))  # prints 90
    find_occurrences()  # 4 occurrences each
    print(sum_divisible_by_3())  # prints 165
    guess_birth_date()
    infinite_series()

File: classTask/Level1/Level1.go
package main

import "fmt"


func main() {
	printHelloWorld()
	printSemicolon()
	printVentures()
	fmt.Println(secondLargest(2, 4, 70, 12, 55, 90, 532, 45, 12)) // prints 90
	findOccurrences() // 4 occurrences each
	fmt.Println(sumDivisibleBy3()) // prints 165
	guessBirthDate() // correct date is 15
	infiniteSeries()
}

// number 0
func printHelloWorld() {
	fmt.Println("Hello World")
}

// number 1
func secondLargest(numbers ...int) int {
	highest := 0
	secondHighest := 0
	for _, num := range numbers {
		if (num > highest) {
			secondHighest = highest
			highest = num
		}
	}
	return secondHighest;
}

// number 2
func printSemicolon() {
	semicolon := "SEMICOLON"
	for _, char := range semicolon {
		fmt.Println(string(char))
	}
}

// number 3
func printVentures() {
	ventures := "VENTURES"
	for i := 0; i < len(ventures); i++ {
		fmt.Printf("%c %c\n", ventures[i], ventures[i+1])
		i++
	}
}

// number 4
func findOccurrences() {
	word := "Mississippi"
	countForS := 0
	countForI := 0
	for _, char := range word {
		if char == 's' {
			countForS++
		} else if char == 'i' {
			countForI++
		}
	}
	fmt.Println("Number of 's':", countForS)
	fmt.Println("Number of 'i':", countForI)
}

// number 5
func sumDivisibleBy3() int {
	sum := 0;
	for i := 1; i <= 30; i++ {
		if (i % 3 == 0) {
			sum += i
		}
	}
	return sum;
}

// number 6
func guessBirthDate() {
	birthday := 15
	var answer int
	for answer != birthday {
		fmt.Print("Enter a guess: ")
		fmt.Scanf("%d", &answer)
		if answer != birthday {
			fmt.Println("Incorrect Guess!")
			fmt.Scanln()
		}
	}
	fmt.Println("Correct Guess!")
}

// number 7
func infiniteSeries() {
	pie := 4.0
	denominator := 3.0
	found := false
	terms := 1
	for !found {
		if terms % 2 == 0 {
			pie += 4.0 / denominator
		} else {
			pie -= 4.0 / denominator
		}
		denominator += 2.0
		fmt.Printf("%-10d%.7f\n", terms, pie)
		terms++

		if fmt.Sprintf("%.7f", pie) == "3.1415900" {
			found = true
		}
	}

	fmt.Println("Pie:", pie)
	fmt.Println("Number of terms:", terms)
}

